### PR TITLE
Dev branch 1.4 starting matplotlib

### PR DESCRIPTION
**Pull request by gecBurton**

### CODE REVIEW

File: backTrader/key_level_strategy.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+from datetime import date, datetime, time, timedelta
+from coinPrice.key_level_finder import range_identification, conn
+
+
+class Key_level_strategy(backtrader.Strategy):
+    #params = dict(
+     #   num_opening_bars=15
+    #)
+    ################################### GETTING KEY LEVELS ############################
+    print("Running Keys Shizzle")
```
Comment (gecBurton) [inline backTrader/key_level_strategy.py:12]: :)

File: backTrader/key_level_strategy.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        '''
+        next loops through each CandleStick data until the end
+        '''
+    def next(self):
+        # print("4")
+        # print("Candle_price {0}".format(self.dataclose[0]))
+        current_bar_datetime = self.data.num2date(self.data.datetime[0])
+        previous_bar_datetime = self.data.num2date(self.data.datetime[-1])
+
+        if 1 != 2:
```
Comment (gecBurton) [inline backTrader/key_level_strategy.py]: this is always true no?

File: backTrader/key_level_strategy.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            self.opening_range_low = self.data.low[0]
+            self.opening_range_high = self.data.high[0]
+            #self.bought_today = False
+
+        # if self.order:
+        #     #print('ORDER ALREADY IN')
+        #     return
+
+        global bought_price  # Question is this the best way to store a value during run time?
```
Comment (gecBurton) [inline backTrader/key_level_strategy.py]: nope, globals should be avoided at all costs. make it an attribute of the class:
```python
    def __init__(self):
        ....
        self.bought_price = None
```

File: backTrader/key_level_strategy.py (hunk 2 above)
Comment (Marl27) [reply backTrader/key_level_strategy.py]: Yes up until the last candle in the DB.

File: backTrader/key_level_strategy.py (hunk 3 above)
Comment (Marl27) [reply backTrader/key_level_strategy.py]: This worked :)

**Pull request merged**